src/magic.cpp

```diff
@@ -405,23 +418,76 @@ int spell::field_intensity() const
                                        type->field_intensity_increment ) ) );
 }
 
+int spell::min_leveled_damage() const
+{
+    return type->min_damage + round( get_level() * type->damage_increment );
+}
+
 int spell::damage() const
 {
-    const int leveled_damage = type->min_damage + round( get_level() * type->damage_increment );
-    if( type->min_damage >= 0 || type->max_damage >= type->min_damage ) {
-        return std::min( leveled_damage, type->max_damage );
-    } else { // if it's negative, min and max work differently
-        return std::max( leveled_damage, type->max_damage );
+    const int leveled_damage = min_leveled_damage();
+
+    if( has_flag( spell_flag::RANDOM_DAMAGE ) ) {
+        return rng( std::min( leveled_damage, type->max_damage ), std::max( leveled_damage,
+                    type->max_damage ) );
+    } else {
+        if( type->min_damage >= 0 || type->max_damage >= type->min_damage ) {
+            return std::min( leveled_damage, type->max_damage );
+        } else { // if it's negative, min and max work differently
+            return std::max( leveled_damage, type->max_damage );
+        }
     }
 }
 
+std::string spell::damage_string() const
+{
+    if( has_flag( spell_flag::RANDOM_DAMAGE ) ) {
+        return string_format( "%d-%d", min_leveled_damage(), type->max_damage );
+    } else {
+        const int dmg = damage();
+        if( dmg >= 0 ) {
+            return string_format( "%d", damage() );
+        } else {
+            return string_format( "+%d", damage() );
+        }
```

Comment:
I think this can be simply
```
return string_format( "%+d", dmg );
```

Reply:
I mean the two `string_format` calls can be merged into one using the `%+d` specifier. Also the current if condition is incorrect, it currently output negative numbers with a leading plus sign.

Reply:
but i don't want the + in the other statement.

Reply:
oh shoot, yeah. i want abs()